### PR TITLE
kubeadm: remove the restriction that the ca.crt can only contain one certificate

### LINKED ISSUE #2628: "kubeadm" does not support full chain in ca.crt
## What keywords did you search in kubeadm issues before filing this one?

"wanted exactly one certificate, got"

## Is this a BUG REPORT or FEATURE REQUEST?

**BUG REPORT**

## Versions

**kubeadm version** (use `kubeadm version`):

```
# kubeadm version
# ./kubeadm version
kubeadm version: &version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.14", GitCommit:"57a3aa3f13699cf3db9c52d228c18db94fa81876", GitTreeState:"clean", BuildDate:"2021-12-15T14:51:22Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Kubernetes version** (use `kubectl version`):
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.18", GitCommit:"6f6ce59dc8fefde25a3ba0ef0047f4ec6662ef24", GitTreeState:"clean", BuildDate:"2021-04-15T03:31:30Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.18", GitCommit:"6f6ce59dc8fefde25a3ba0ef0047f4ec6662ef24", GitTreeState:"clean", BuildDate:"2021-04-15T03:23:41Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```
- **Cloud provider or hardware configuration**: 
Not relying on any cloud providers. 

- **OS** (e.g. from /etc/os-release):
```
# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
VERSION_CODENAME=stretch
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

- **Kernel** (e.g. `uname -a`):
```
Linux kernel 5.4.56
```

- **Others**:

## What happened?

This issue is a follow-up issue from https://github.com/kubernetes/kubeadm/issues/2618

We are in the middle of migrating existing kubeadm provisioned k8s PKI, on our cluster, to an external CA (pki).  And as a part of that we lay an intermediate CA on the filesystem, which has the full chain.

But one of the things that we noticed is when we perform kubeadm commands related to handling kubernetes certificates, it fails with this error which indicates that it doesn't like full chain

https://github.com/kubernetes/kubernetes/blob/v1.20.14/cmd/kubeadm/app/phases/certs/renewal/readwriter.go#L149

```
# kubeadm certs check-expiration  --v=7

I1217 21:31:25.171210 1318693 loader.go:379] Config loaded from file:  /etc/kubernetes/admin.conf
I1217 21:31:25.172054 1318693 cert_rotation.go:137] Starting client certificate rotation controller
[check-expiration] Reading configuration from the cluster...
[check-expiration] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
I1217 21:31:25.174964 1318693 round_trippers.go:422] GET https://k8s-master.mycompany.org:6443/api/v1/namespaces/kube-system/configmaps/kubeadm-config?timeout=10s
I1217 21:31:25.174984 1318693 round_trippers.go:429] Request Headers:
I1217 21:31:25.174995 1318693 round_trippers.go:433]     Accept: application/json, */*
I1217 21:31:25.175004 1318693 round_trippers.go:433]     User-Agent: kubeadm/v1.20.14 (linux/amd64) kubernetes/57a3aa3
I1217 21:31:25.187338 1318693 round_trippers.go:448] Response Status: 200 OK in 12 milliseconds
I1217 21:31:25.188554 1318693 round_trippers.go:422] GET https://k8s-master.mycompany.org:6443/api/v1/namespaces/kube-system/configmaps/kube-proxy?timeout=10s
I1217 21:31:25.188583 1318693 round_trippers.go:429] Request Headers:
I1217 21:31:25.188589 1318693 round_trippers.go:433]     Accept: application/json, */*
I1217 21:31:25.188594 1318693 round_trippers.go:433]     User-Agent: kubeadm/v1.20.14 (linux/amd64) kubernetes/57a3aa3
I1217 21:31:25.190974 1318693 round_trippers.go:448] Response Status: 200 OK in 2 milliseconds
I1217 21:31:25.192035 1318693 round_trippers.go:422] GET https://k8s-master.mycompany.org:6443/api/v1/namespaces/kube-system/configmaps/kubelet-config-1.18?timeout=10s
I1217 21:31:25.192050 1318693 round_trippers.go:429] Request Headers:
I1217 21:31:25.192056 1318693 round_trippers.go:433]     Accept: application/json, */*
I1217 21:31:25.192061 1318693 round_trippers.go:433]     User-Agent: kubeadm/v1.20.14 (linux/amd64) kubernetes/57a3aa3
I1217 21:31:25.194239 1318693 round_trippers.go:448] Response Status: 200 OK in 2 milliseconds
I1217 21:31:25.195557 1318693 loader.go:379] Config loaded from file:  /etc/kubernetes/kubelet.conf
I1217 21:31:25.195799 1318693 round_trippers.go:422] GET https://k8s-master.mycompany.org:6443/api/v1/nodes/n231-029-107?timeout=10s
I1217 21:31:25.195814 1318693 round_trippers.go:429] Request Headers:
I1217 21:31:25.195822 1318693 round_trippers.go:433]     User-Agent: kubeadm/v1.20.14 (linux/amd64) kubernetes/57a3aa3
I1217 21:31:25.195829 1318693 round_trippers.go:433]     Accept: application/json, */*
I1217 21:31:25.198132 1318693 round_trippers.go:448] Response Status: 200 OK in 2 milliseconds
I1217 21:31:25.199490 1318693 round_trippers.go:422] GET https://k8s-master.mycompany.org:6443/api/v1/namespaces/kube-system/pods?fieldSelector=spec.nodeName%3Dn231-029-107&labelSelector=component%3Dkube-apiserver%2Ctier%3Dcontrol-plane
I1217 21:31:25.199510 1318693 round_trippers.go:429] Request Headers:
I1217 21:31:25.199515 1318693 round_trippers.go:433]     Accept: application/json, */*
I1217 21:31:25.199520 1318693 round_trippers.go:433]     User-Agent: kubeadm/v1.20.14 (linux/amd64) kubernetes/57a3aa3
I1217 21:31:25.203507 1318693 round_trippers.go:448] Response Status: 200 OK in 3 milliseconds
[check-expiration] Error reading configuration from the Cluster. Falling back to default configuration

I1217 21:31:25.305491 1318693 interface.go:400] Looking for default routes with IPv4 addresses
I1217 21:31:25.305503 1318693 interface.go:405] Default route transits interface "eth0"
I1217 21:31:25.305657 1318693 interface.go:208] Interface eth0 is up
I1217 21:31:25.305709 1318693 interface.go:256] Interface "eth0" has 3 addresses :[10.231.29.107/21 fdbd:dc61:ff::231:29:107/128 fe80::f816:3eff:fe94:fc2b/64].
I1217 21:31:25.305726 1318693 interface.go:223] Checking addr  10.231.29.107/21.
I1217 21:31:25.305734 1318693 interface.go:230] IP found 10.231.29.107
I1217 21:31:25.305740 1318693 interface.go:262] Found valid IPv4 address 10.231.29.107 for interface "eth0".
I1217 21:31:25.305746 1318693 interface.go:411] Found active IP 10.231.29.107
I1217 21:31:25.309910 1318693 certs.go:506] validating certificate period for CA certificate
I1217 21:31:25.310233 1318693 certs.go:506] validating certificate period for API server certificate
I1217 21:31:25.310815 1318693 certs.go:506] validating certificate period for API server kubelet client certificate
I1217 21:31:25.311314 1318693 loader.go:379] Config loaded from file:  /etc/kubernetes/admin.conf
wanted exactly one certificate, got 4
k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/renewal.(*kubeConfigReadWriter).Read
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/renewal/readwriter.go:149
k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/renewal.(*Manager).GetCertificateExpirationInfo
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/renewal/manager.go:318
k8s.io/kubernetes/cmd/kubeadm/app/cmd/alpha.newCmdCertsExpiration.func1
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/alpha/certs.go:404
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:850
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:958
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:895
k8s.io/kubernetes/cmd/kubeadm/app.Run
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:50
main.main
        _output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:25
runtime.main
        /usr/local/go/src/runtime/proc.go:204
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1374
```

 
The full chain of certificates in the intermediate CA

```
# openssl crl2pkcs7 -nocrl -certfile /etc/kubernetes/pki/ca.crt | openssl pkcs7 -print_certs  -noout | grep "subject=" | wc -l

4

```

## What you expected to happen?

`kubeadm certs check-expiration` should emit output as expected with EXTERNALLY MANAGED, as yes 

It should honor full chain in the intermediate CA.crt

## How to reproduce it (as minimally and precisely as possible)?

Create a cluster with using kubeadm generated PKI and rotate the CA manually without the `apiserver.key`, `ca.key` and `etcd/ca.key` and issue all the required certs. Restart all the components gracefully to complete the migration. 

Execute the  `kubeadm certs check-expiration` to see the error.

## Anything else we need to know?

This is follow-up of issues related to https://github.com/kubernetes/kubeadm/issues/2618

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
kubeadm: remove the restriction that the `ca.crt` can only contain one certificate.
If there is more than one certificate, kubeadm will pick the first one by default.

#### Which issue(s) this PR fixes:

Fixes kubernetes/kubeadm#2628

#### Does this PR introduce a user-facing change?

```release-note
kubeadm: remove the restriction that the ca.crt can only contain one certificate. If there is more than one certificate in the ca.crt file, kubeadm will pick the first one by default.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by pacoxu — APPROVED** [review]
/lgtm
/cc @neolit123

**Comment by SataQiu** [comment]
/hold 
for more review @neolit123 @alwaysastudent

**Review by neolit123 — COMMENTED** [review]
/lgtm

thanks, we can also backport the change to the older versions in support - 1.20+

**Comment by SataQiu** [comment]
> thanks, we can also backport the change to the older versions in support - 1.20+

/hold cancel

Will do cherry-pick after this PR is merged.

**Pull request merged**
